lib/pages/new_private_chat/widget/expansion_list.dart

```diff
@@ -23,171 +19,138 @@ class ExpansionList extends StatelessWidget {
-          if (success.data.isEmpty) {
-            return Column(
-              crossAxisAlignment: CrossAxisAlignment.start,
-              children: [
-                const SizedBox(
-                  height: 12,
-                ),
-                NoContactsFound(keyword: success.keyword),
-                _MoreListTile(),
-                _NewGroupButton(
-                  onPressed: goToNewGroupChat,
-                ),
-                _GetHelpButton(),
-              ],
-            );
-          }
-
-          final isSearchEmpty = success.keyword.isEmpty;
-          final expansionList = [
-            const SizedBox(
-              height: 4,
-            ),
-            _buildTitle(context, success.data.length),
-            ValueListenableBuilder<bool>(
-              valueListenable: isShowContactsNotifier,
-              builder: ((context, isShow, child) {
-                if (!isShow) {
-                  return const SizedBox.shrink();
-                }
-                return ListView.builder(
-                  shrinkWrap: true,
-                  physics: const NeverScrollableScrollPhysics(),
-                  itemCount: success.data.length,
-                  itemBuilder: (context, index) {
-                    final contact = success.data[index];
-                    return InkWell(
-                      onTap: () {
-                        onContactTap(
-                          context,
-                          contact,
-                        );
-                      },
-                      borderRadius: BorderRadius.circular(16.0),
-                      child: ExpansionContactListTile(
-                        contact: contact,
-                        highlightKeyword: success.keyword,
-                      ),
-                    );
-                  },
-                );
-              }),
-            ),
-            ValueListenableBuilder(
-              valueListenable: isSearchModeNotifier,
-              builder: (context, isSearchMode, child) {
-                if (isSearchMode) {
-                  return const SizedBox.shrink();
-                }
-                return ValueListenableBuilder(
-                  valueListenable: isShowContactsNotifier,
-                  builder: (context, isShow, child) {
-                    if (!isShow || success.isEnd) {
-                      return const SizedBox.shrink();
-                    }
-                    return const Padding(
-                      padding: EdgeInsets.all(16.0),
-                      child: Center(
-                        child: CircularProgressIndicator(),
-                      ),
-                    );
-                  },
-                );
-              },
-            ),
-          ];
+      valueListenable: presentationContactsNotifier,
+      builder: (context, presentationContacts, child) {
+        final isSearchModeEnable =
+            textEditingController.text.isNotEmpty && isSearchModeNotifier.value;
```

Comment:
isSearchModeNotifier maybe noneed, because you havent listen it anywhere

Reply:
@nqhhdev  you can remove isSearchModeNotifier
`final isSearchMode = textEditingController.text.isNotEmpty;`